velox/exec/RowContainer.cpp

```diff
@@ -537,6 +539,45 @@ void RowContainer::store(
   }
 }
 
+void RowContainer::storeVector(
+    const DecodedVector& decoded,
+    vector_size_t size,
+    char* const* rows,
+    int32_t column) {
+  auto numKeys = keyTypes_.size();
+  bool isKey = column < numKeys;
+  updateColumnMayHaveNulls(column, decoded.mayHaveNullsRecursive());
+  if ((isKey && !nullableKeys_) || !decoded.mayHaveNullsRecursive()) {
+    for (int i = 0; i < size; ++i) {
+      VELOX_DYNAMIC_TYPE_DISPATCH(
```

Comment:
It seems inefficient to do type dispatch at every loop iteration. Can this be done outside of the loop?

Reply:
Sure, updated thanks.